### DISCUSSION TITLE
iOS - Share PDFs to Paperparrot

### DISCUSSION THREAD

**Discussion by AlexGeek77** [LeoKlaus/Paperparrot#172] [Feature Requests]
Similar to how you have the MacOS with a share extension that allows documents to be sent to Paperparrot it would be nice to have the same functionality in iOS devices. So when I browse a website that has a PDF I can share it directly to Paperparrot and thus paperless-ngx.

**Comment by LeoKlaus** [comment]
The iOS version has a share extension too.
It should show up for all supported file types (pdf, images, office documents and eml).

Maybe check if the share extension is disabled in the share sheet (scroll to right in the available apps and tap more, then edit) or restart your device.

**Comment by AlexGeek77** [comment]
Ok, it took a reboot of my phone to get it to show up properly.  Thanks for that recommendation.

Weirdly, just like on my Macbook it shows the CSRF cookie error. I started to go through the same steps I had to do on my Mac to get it working. When I used the local IP address it worked. Then I got an error when I used the Tailscale name:
![17978474-DA18-4DE3-B799-62E660C8C050_4_5005_c](https://github.com/LeoKlaus/Paperparrot/assets/1232415/c613223c-e867-445b-ba35-6bb2501f35fe)
So then I tried to remove the local IP address and it told me this:
![9FA7B203-0761-4F51-9C20-D600521D8FD3_4_5005_c](https://github.com/LeoKlaus/Paperparrot/assets/1232415/7273d445-e03f-4694-b7d9-c80b0a4cb229)
I deleted the app and reinstalled the app. I added the FQDN Tailscale address but it still didn't like that so I added the regular Tailscale name and deleted the FQDN Tailscale name successfully this time. After that the iOS app is working properly for sharing.

I'll say that all the server names I was using worked just fine in the iOS app, it's just the Share extension that was bugging out. All good now for me. Ultimately I think it's some issue with how it's handling the Tailscale network naming since it works fine with local direct IP address. I know on the Mac it had issues when I used the Tailscale IP address too though. So there's something weird going on with the response happening in the Share extension that's different than the App natively.

**Reply by LeoKlaus** [reply]
That's weird. Thanks for the update.

I have to admit, I find Tailscales behavior a bit weird sometimes, one thing that threw me off was that (at least for the instance I spun up for testing) the devices in my Tailscale network did _not_ receive IPs in any of the traditional local ranges (i.e. 192.168. or 10.). This causes iOS to treat those devices as remote, which in turn means HTTPS is enforced by default.

There's still something wrong with the share extension I assume, but the entitlements should be correct and it (too) only uses default URLSessions.
